Mathlib/Combinatorics/SimpleGraph/Connectivity.lean

```diff
@@ -2064,6 +2067,24 @@ lemma mem_coe_supp_of_adj {v w : V} {H : Subgraph G} {c : ConnectedComponent H.c
 
 end ConnectedComponent
 
+lemma pairwise_disjoint_supp_connectedComponent (G : SimpleGraph V) :
+    Pairwise fun c c' : ConnectedComponent G ↦ Disjoint c.supp c'.supp := by
+  intro _ _ h s hsx hsy
+  simp only [Set.toFinset_card, Finset.mem_univ, ne_eq, Set.le_eq_subset,
+    Set.bot_eq_empty, Set.subset_empty_iff, Set.eq_empty_iff_forall_not_mem] at *
+  intro v hv
+  have hsxv := hsx hv
+  have hsyv := hsy hv
+  rw [ConnectedComponent.mem_supp_iff] at *
+  rw [hsxv] at hsyv
+  exact h hsyv
+
+lemma iUnion_connectedComponentSupp (G : SimpleGraph V) :
+    ⋃ c : G.ConnectedComponent, c.supp = Set.univ := by
+  refine Set.eq_univ_of_forall fun v ↦ ⟨G.connectedComponentMk v, ?_⟩
+  simp only [Set.mem_range, SetLike.mem_coe]
+  exact ⟨by use G.connectedComponentMk v; exact rfl, rfl⟩
```

Comment:
Ideally these two lemmas should be proven about arbitrary equivalence relations. That might be outside the scope of this PR, but maybe leave a comment?

Reply:
Added TODOs